scylla-cql/src/frame/value.rs

```diff
@@ -230,6 +215,80 @@ impl<'a> Iterator for SerializedValuesIterator<'a> {
     }
 }
 
+/// Represents List of ValueList for Batch statement
+///
+/// This trait is not implemented directly, but rather implemented through `BatchValuesGatWorkaround`
+/// (until GATs are made available in Rust)
+pub trait BatchValues: for<'r> BatchValuesGatWorkaround<'r> {}
+impl<T: for<'r> BatchValuesGatWorkaround<'r> + ?Sized> BatchValues for T {}
+
+#[allow(clippy::len_without_is_empty)]
+pub trait BatchValuesGatWorkaround<'r> {
+    type BatchValuesIter: BatchValuesIterator;
+    fn batch_values_iter(&'r self) -> Self::BatchValuesIter;
+    fn len(&'r self) -> usize;
+}
+
+/// An iterator-like for `ValueList`
+///
+/// An instance of this can be easily obtained from `IT: Iterator<Item: ValueList>`: that would be
+/// `BatchValuesIteratorFromIterator<IT>`
+///
+/// It's just essentially making methods from `ValueList` accessible instead of being an actual iterator because of
+/// several compiler limitations that would otherwise be very complex to overcome.  
```

Comment:
There are two trailing spaces at the end of this line. Please replace them with a backslash `\`.
We decided to use backslashes for newlines in documentation because they are visible without special tools and some people have plugins that automatically remove trailing whitespace characters.